app/lib/three_scale/representer.rb

```diff
@@ -21,4 +22,8 @@ def wraps_resource(root = true)
     end
   end
   extend Wrapping
+
+  def representable_map(options, format)
```

Comment:
This change is quite confusing...

But this is basically a copy from https://github.com/trailblazer/representable/blob/v3.0.4/lib/representable.rb#L63-L65

I added it because otherwise, the execution ended up using this method here: https://github.com/trailblazer/representable/blob/v3.0.4/lib/representable/cached.rb#L18-L20

And it was just using a wrong binding... Unfortunately, I don't remember the exact error anymore :sweat_smile:

Reply:
You can see the tests that fail in https://app.circleci.com/pipelines/github/3scale/porta/27844/workflows/dbd93801-7760-49a4-8d37-46d32ca11aeb

Reply:
I'm surprised with this, it looks like a bug in the gem or something. I mean, I assume [here](https://github.com/trailblazer/representable/blob/3b66945c7de584a0c70dc1316101903f7ad1ebb7/lib/representable.rb#L70) they intend to call [this](https://github.com/trailblazer/representable/blob/v3.0.4/lib/representable.rb#L63-L65) `representable_map`, however they include `representable/cached` which apparently overrides `representable_map`. Then, why they define the original `representable_map` in the first place? How could it ever get called? And how is it possible nobody noticed this before?

Reply:
Yeah, I don't understand it either...

This commit message is interesting https://github.com/trailblazer/representable/commit/b26aec4bbfe27d7c68526efd9d00c78980ed6bd4

> ALWAYS use Cached in Decorator. this might break things when people later include the format engine module, or include multiple of the latter. please contact me then.

"when people later include the format engine module" - include where? what is the "format engine module"?... I don't get it, but maybe we need to contact the guy? :laughing:

Reply:
> but maybe we need to contact the guy?

Maybe, or open an issue in the repo so they can clarify this for everybody.

Reply:
OK, I think I understood what this is about, and I think I've applied a proper fix now: https://github.com/3scale/porta/pull/3725/commits/6eee8c3171d5351b6ddc9d071830a835ba6bd963

Reply:
@jlledom I looked at it again, and the "proper fix" is not proper actually :sweat_smile: 

So, the initial problem was that the XML and JSON bindings were used incorrectly because of caching, and serializing a JSON was causing `document must be a Nokogiri::XML::Node`.

After the "fix" the caching was still in place, and so the generic Hash binding was used, because it was cached on the representer initialization. Which does not crash, but does cause a wrongly formatted XML.

I actually found this issue, which is example the same thing we're dealing with: https://github.com/trailblazer/representable/issues/180

And the response of the maintainer is: https://github.com/trailblazer/representable/issues/180#issuecomment-169881369

So, I returned to the original solution, and added more comments to explain what this change does.